modalapi/wifi.py

```diff
@@ -117,12 +117,46 @@ def enable_hotspot(self):
         try:
             subprocess.check_output(['sudo', 'systemctl', 'enable', 'wifi-hotspot']).strip().decode('utf-8')
             subprocess.check_output(['sudo', 'systemctl', 'start', 'wifi-hotspot']).strip().decode('utf-8')
+            self.wifi_status = True
         except:
             logging.debug('Wifi hotspot enabling failed')
+            self.wifi_status = False
 
     def disable_hotspot(self):
         try:
             subprocess.check_output(['sudo', 'systemctl', 'stop', 'wifi-hotspot']).strip().decode('utf-8')
             subprocess.check_output(['sudo', 'systemctl', 'disable', 'wifi-hotspot']).strip().decode('utf-8')
+            self.wifi_status = False
         except:
             logging.debug('Wifi hotspot disabling failed')
+            self.wifi_status = True
+
+    def configure_wifi(self, ssid, password):
+        config_lines = [
+            'ctrl_interface=DIR=/var/run/wpa_suplicant GROUP=netdev',
+            'update_config=1',
+            'country=US',
+            '\n'
+            'network={',
+            '\tssid="{}"'.format(ssid),
+            '\tpsk="{}"'.format(password),
+            '}'
+        ]
+        config = '\n'.join(config_lines)
+
+        os.popen("sudo chmod a+w /etc/wpa_supplicant/wpa_supplicant.conf")
+
+        with open("/etc/wpa_supplicant/wpa_supplicant.conf", "w") as wifi:
+            wifi.write(config)
+        logging.info("Wifi config added. Refreshing config")
+
+        os.popen("sudo wpa_cli -i wlan0 reconfigure")
+
+    def get_wifi_credentials(self, file_path):
+        file_path = '/etc/wpa_supplicant/wpa_supplicant.conf'
+        with open(file_path, 'r') as file:
+            for line in file:
+                if 'ssid=' in line:
+                    current_ssid = line.split('"')[1]
```

Comment:
I feel like this line could fail if the file were hand edited.  For example, I believe single ticks around strings is still valid.  At a minimum, it should not fail with a "list index out of range" if the string can't be parsed.